itur/models/itu1623.py

```diff
@@ -0,0 +1,668 @@
+
+        # Step 2:   Calculate STD of the conditional fade slope
+        # s is a parameter which depends on climate and elevation angle; an
+        # overall average value in Europe and the United States of America,
+        # at elevations between 10° and 50°, is s = 0.01
+        s = 0.01
+        sigma_z = s * F * A
+
+        # Step 3a:  Calculate the conditional probability
+        p = 2 / (np.pi * sigma_z * (1 + (z / (sigma_z) ** 2) ** 2))  # eq. 20
+
+        # Step 3b : If required, calculate p(ζ |A), the conditional
+        # probability (complementary cumulative distribution function) that
+        # the fade slope ζ is exceeded for a given attenuation value, A:
+        z_over_sigmaz = z / sigma_z
+        abs_z_over_sigmaz = np.abs(z) / sigma_z
+        P = (
+            0.5
+            - z_over_sigmaz / (np.pi * (1 + z_over_sigmaz ** 2))
+            - (np.arctan(z_over_sigmaz) / np.pi)
+        )
+
+        # Step 3b2 : calculate p(ζ |A), the conditional probability that the
+        # absolute value of the fade slope ζ is exceeded for a given
+        # attenuation value, A:
+        P2 = (
+            1
+            - 2 * abs_z_over_sigmaz / (np.pi * (1 + abs_z_over_sigmaz ** 2))
+            - (2 * np.arctan(abs_z_over_sigmaz) / np.pi)
+        )
+
+        return p, P, P2, sigma_z
+
+    @classmethod
+    def fade_depth(self, N_target, D_target, A, PofA, el, f):
+
+        d_target = np.atleast_1d(D_target)
+
+        def delta_N_events(x):
+            P_it = 10 ** (np.interp(x, A, np.log10(PofA)))
+            T_tot_it = (P_it / 100) * 365.25 * 86400
+            _, _, N_it, _ = self.fade_duration(d_target, x, el, f, T_tot_it)
+            delta = N_target - N_it
+            return delta
+
+        a_min = fsolve(delta_N_events, 1)  # a_min should have shape (1,)
+        return a_min.item()
+
+
+class _ITU1623_0_:
+    def __init__(self):
+        self.__version__ = 0
+        self.year = 2003
+        self.month = 4
+        self.link = "https://www.itu.int/rec/R-REC-P.1623-0-200304-S/en"
+
+    @staticmethod
+    def fade_duration(*args, **kwargs):
+        return _ITU1623_1_.fade_duration(*args, **kwargs)
+
+    @staticmethod
+    def fade_slope(*args, **kwargs):
+        return _ITU1623_1_.fade_slope(*args, **kwargs)
+
+    @staticmethod
+    def fade_depth(*args, **kwargs):
+        return _ITU1623_1_.fade_depth(*args, **kwargs)
+
+
+__model = __ITU1623__()
+
+
+def change_version(new_version):
+    """
+    Change the version of the ITU-R P.1623 recommendation currently being used.
+
+    This function changes the model used for the ITU-R P.1623 recommendation
+    to a different version.
+
+    Parameters
+    ----------
+    new_version : int
+        Number of the version to use.
+        Valid values are:
+          * 1: Activates recommendation ITU-R P.1623-1 (03/2005) (Current version)
+          * 0: Activates recommendation ITU-R P.1623-0 (04/2003) (Superseded)
+    """
+    global __model
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Global variables are dangerous because they can be simultaneously accessed from multiple sections of a program. This frequently results in bugs. Most bugs involving global variables arise from one function reading and acting on the value of a global variable before another function has the chance to set it to an appropriate value. We recommend using a configuration module to mutate global state.

[Learn more](https://docs.python.org/3/faq/programming.html#how-do-i-share-global-variables-across-modules)